core/core.php

```diff
@@ -181,4 +181,86 @@ public function validate_badge_order($user_id)
 
 		return true;
 	}
+
+	/**
+	 * Gets the url to the user avatar.
+	 * Returns empty string on failure or if there is no avatar.
+	 *
+	 * @param array $user_data Avatar data from the user table
+	 * @return array
+	 */
+	public function get_avatar_data($user_data)
+	{
+		global $user, $config;
+		global $phpbb_container, $phpbb_dispatcher;
```

Comment:
There's no need to hook into the current `user` object here since we do not care about the logged in user. `config` can be injected into the `phpbb_container` in the class constructor. `phpbb_dispatcher` is not needed. So none of these globals is really needed.

Reply:
I removed the unused globals and moved the rest to the constructor.